src/lib/libstrings.js

```diff
@@ -179,23 +180,17 @@ addToLibrary({
     // Always use TextEncoder when TEXTENCODER == 2
     var encoded = UTF8Encoder.encode(str);
     var bytesToWrite = Math.min(encoded.length, maxBytesToWrite - 1); // -1 for null terminator
-    
-    for (var i = 0; i < bytesToWrite; ++i) {
-      heap[outIdx + i] = encoded[i];
-    }
-    // Null-terminate the string
+    encoded = encoded.subarray(0, bytesToWrite)
+    heap.set(encoded, outIdx)
     heap[outIdx + bytesToWrite] = 0;
     return bytesToWrite;
 #else
     // When using conditional TextEncoder, use it for longer strings if available
     if (str.length > 16 && UTF8Encoder) {
       var encoded = UTF8Encoder.encode(str);
```

Comment:
We could also consider using `encodeInto` using a `subarray` of the heap?

The downside I think is that `envodeInto` was added a little later so might be available in fewer places.

Reply:
> We could also consider using `encodeInto` using a `subarray` of the heap?

That is a fine idea. I think it's good to work on as a separate PR, if someone wants to. Along with the appropriate `#if MIN_SAFARI_VERSION >= 140500` gates etc.